drivers/counter/counter_gecko_stimer.c

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright (c) 2021, Sateesh Kotapati
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT silabs_gecko_rtcc
```

Comment:
I do not think that a different driver with the same compatibility string is a good idea. As far as i understand both drivers counter_gecko_rtcc.c and counter_gecko_stimer.c use the same hardware peripheral and it is just a different driver implementation.

Reply:
Created a different compatibility string.